node/node.go

```diff
@@ -576,6 +557,23 @@ func (n *Node) createMetaChainBootstrapper(rounder consensus.Rounder) (process.B
 	return bootstrap, nil
 }
 
+func (n *Node) createMiniblocksGetter() (process.MiniBlockDataGetter, error) {
+	if check.IfNil(n.dataPool) {
+		return nil, process.ErrNilPoolsHolder
+	}
+	if check.IfNil(n.store) {
+		return nil, process.ErrNilStorage
+	}
+
+	arg := getters.ArgMiniBlockDataGetter{
+		MiniBlockPool:    n.dataPool.MiniBlocks(),
+		MiniBlockStorage: n.store.GetStorer(dataRetriever.MiniBlockUnit),
+		Marshalizer:      n.internalMarshalizer,
+	}
+
+	return getters.NewMiniBlockDataGetter(arg)
```

Comment:
Agree, but since all required components are there, and for a bug-less future use I have chosen this way. StartConsensus needs to be refactored anyway: the name is misleading, it does too many things, should have called just some factory methods.